packages/backend/src/apps/m365-excel/actions/create-table-row/index.ts

```diff
@@ -0,0 +1,174 @@
+import type { IJSONObject, IRawAction } from '@plumber/types'
+
+import WorkbookSession from '../../common/workbook-session'
+
+interface TableInfo {
+  columnCount: number
+  rowIndex: number
+}
+
+interface ColumnValue {
+  index: number
+  value: string
+}
+
+// MS Graph expects the row data to be an array of the same length as the number
+// of columns in the table, with values in the same order as the corresponding
+// column index.
+function constructMsGraphArgment(
+  tableInfo: TableInfo,
+  columnValues: ColumnValue[],
+): Array<string | null> {
+  columnValues.sort((lhs, rhs) => lhs.index - rhs.index)
+  const largestColumnIndex = columnValues[columnValues.length - 1].index
+
+  if (tableInfo.columnCount <= largestColumnIndex) {
+    throw new Error('Trying to write to non-existent column.')
+  }
+
+  const result: Array<string | null> = new Array(tableInfo.columnCount).fill(
+    null,
+  )
+  for (const columnValue of columnValues) {
+    result[columnValue.index] = columnValue.value
+  }
+  return result
+}
+
+const action: IRawAction = {
+  name: 'Create row',
+  key: 'create-table-row',
```

Comment:
nit: this should be camelCase?

Reply:
yeah why u no stick to norm

Reply:
old man forgot. Paiseh. Cos app keys are dash case I think.